app/views/lessons/project_submissions/index.html.erb

```diff
@@ -2,55 +2,67 @@
-          <% end %>
-        </div>
       </header>
 
-      <%= turbo_frame_tag 'submissions-list', data: { test_id: 'submissions-list', controller: 'sort' } do %>
-        <%= render ProjectSubmissions::ItemComponent.new(project_submission: @current_user_submission, current_user:) do |component| %>
-          <%= component.with_title(title: current_user.username, url: dashboard_path) %>
-        <% end %>
+      <div class="py-4 flex justify-end">
+          <div class="w-64">
+            <%= render SortComponent.new(
+                  selected: { value: params[:sort], direction: params[:direction] },
+                  data_attributes: { turbo_frame: 'submissions-list' },
+                  options: [
+                    { value: 'created_at', label: 'Newest', direction: 'desc', default: true },
+                    { value: 'created_at', label: 'Oldest', direction: 'asc', default: false },
+                    { value: 'likes_count', label: 'Most liked', direction: 'desc', default: false }
+                  ]
+                ) %>
+          </div>
+      </div>
 
-        <% @project_submissions.each do |project_submission| %>
-          <%= render ProjectSubmissions::ItemComponent.new(project_submission:, current_user:) do |component| %>
-            <%= component.with_title(title: project_submission.user.username) %>
-          <% end %>
-        <% end %>
+      <%= render CardComponent.new do |card| %>
+        <% card.with_body(classes: 'py-4 px-4') do %>
+          <% if @project_submissions.any? %>
+
+            <div class="divide-y">
+              <% @project_submissions.each do |project_submission| %>
+                <%= render ProjectSubmissions::ItemComponent.new(project_submission:) do |component| %>
+                  <%= component.with_title(title: project_submission.user.username) %>
+                <% end %>
+              <% end %>
+            </div>
 
-        <div class="p-8 pb-16 hidden only:block">
-          <div class="text-center">
-            <%= inline_svg_tag 'icons/folder-plus.svg', class: 'mx-auto h-12 w-12 text-gray-400 dark:text-gray-500', aria: true, title: 'No solutions', desc: 'No solutions icon' %>
+          <% else %>
+            <div class="p-8">
+              <div class="text-center">
+                <%= inline_svg_tag 'icons/folder-plus.svg', class: 'mx-auto h-12 w-12 text-gray-400 dark:text-gray-500', aria: true, title: 'No solutions', desc: 'No solutions icon' %>
 
-            <h3 class="mt-2 text-xl font-semibold text-gray-900 dark:text-gray-200">No solutions</h3>
-            <p class="mt-1 text-sm text-gray-500 dark:text-gray-400">Be the first</p>
+              <h3 class="mt-2 text-xl font-semibold text-gray-900 dark:text-gray-200">No solutions</h3>
+              <%= link_to 'Submit your solution', lesson_path(@lesson, anchor: 'project-solution'), class: 'mt-2 text-sm button button--secondary py-2', data: { turbo_frame: '_top'} %>
```

Comment:
It feels a bit janky to navigate them back to the lesson page to submit their solution so they effectively have to click twice on the same text to submit their solution. Couldn't we use turbo's power here to just load the solution component in place? Maybe one for a future PR

Reply:
Good call out - yeah this was a bit of a cop out. The new user solution component being different from the item component we use on this page makes it a little bit more difficult to create a solution here. But I'll have a think about ways around that.

In the meantime I think we're probably better off prompting users to return to the project page instead of submitting. That feels a bit less janky.